vslib/src/sai_vs_vlan.cpp

```diff
@@ -78,10 +161,23 @@ sai_status_t vs_remove_vlan_member(
 
     SWSS_LOG_ENTER();
 
-    return meta_sai_remove_oid(
+    sai_vlan_id_t vlan_id = get_vlan_id_from_member(vlan_member_id);
+
+    sai_status_t status = meta_sai_remove_oid(
             SAI_OBJECT_TYPE_VLAN_MEMBER,
             vlan_member_id,
             &vs_generic_remove);
+
+    if (status == SAI_STATUS_SUCCESS)
+    {
+        auto it = vlan_members_map[vlan_id].find(vlan_member_id);
+
+        vlan_members_map[vlan_id].erase(it);
```

Comment:
no, as you can se its only executed after success, so vlan member id will exists